comparison/loadtest.go

```diff
@@ -216,12 +222,40 @@ func initLoadTest(t *terraform.Terraform, buildCfg BuildConfig, dumpFilename str
 		return fmt.Errorf("invalid db engine %s", dpConfig.TerraformDBSettings.InstanceEngine)
 	}
 
+	resetBucketCmds := []localCmd{}
+	if s3BucketURI != "" && tfOutput.HasS3Bucket() {
+		deleteBucketCmd := localCmd{
+			msg:   "Emptying S3 bucket",
+			value: []string{"--profile", t.Config().AWSProfile, "s3", "rm", "s3://" + tfOutput.S3Bucket.Id, "--recursive"},
+		}
+
+		prepopulateBucketCmd := localCmd{
+			msg:   "Pre-populating S3 bucket",
+			value: []string{"--profile", t.Config().AWSProfile, "s3", "cp", s3BucketURI, "s3://" + tfOutput.S3Bucket.Id, "--recursive"},
+		}
+
+		resetBucketCmds = []localCmd{deleteBucketCmd, prepopulateBucketCmd}
+	}
+
 	if dumpFilename == "" {
 		cmds = append(cmds, startCmd, createAdminCmd, initDataCmd)
 	} else {
 		cmds = append(cmds, loadDBDumpCmd, startCmd)
 	}
 
+	// Resetting the buckets can happen concurrently with the rest of the remote commands
+	resetBucketErrCh := make(chan error)
+	go func() {
+		for _, c := range resetBucketCmds {
+			mlog.Info(c.msg)
+			if err := exec.Command("aws", c.value...).Run(); err != nil {
+				resetBucketErrCh <- fmt.Errorf("failed to run local cmd %q: %w", c.value, err)
+				return
+			}
+		}
+		resetBucketErrCh <- nil
+	}()
+
```

Comment:
One thing to note, if we receive from `cancelCh`, we immediately exit from this function. We are not cancelling these commands and it can leak the commands. Wondering if we should also cancel them by using https://pkg.go.dev/os/exec#CommandContext

Reply:
Good catch! Please, check https://github.com/mattermost/mattermost-load-test-ng/pull/524/commits/a0711b21d19d8e743343beacd3ca93f7f3e7ab44 :pray: